UITests/XCUIElement+Utils.swift

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2017-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import XCTest
+
+extension XCUIElement {
+    
+    var stringValue: String? {
+        value as? String
+    }
+    
+    func waitForText(predicate: @escaping (String?) -> Bool, timeout: TimeInterval) -> Bool {
```

Comment:
Is there any particular reason you chose to use semaphores, instead of XCTestExpectations?

Reply:
FYI you can use expectations that are evaluated with predicates/blocks.  For example, here's an expectation I wrote recently in another project:

```
        let waiter = expectation(for: .init(format: "needsUpdate == 0"),
                                  evaluatedWith: configs) { () -> Bool in
             return !configs.needsUpdate
         }
         wait(for: [waiter], timeout: timeout)
```

Reply:
You made me think about that. So I refactored and used `XCTNSPredicateExpectation`. There're only 2 lines of code. Simpler and easier to maintain now.